backend/src/api/routes/users.ts

```diff
@@ -0,0 +1,84 @@
+import express, { NextFunction } from 'express';
+import { extractPayload, Company, Query, CompanyUser, instanceOfCompany, instanceOfCompanyUser } from '../../utils';
+import { Fields } from 'formidable';
+import { genereateAdminConnection, getPoolClient, executeTransaction } from '../../services/databasefunctions';
+import { logger } from '../../services/logger';
+import { PoolClient, Pool } from 'pg';
+const app: express.Application = express();
+
+const newCompany = async (req: express.Request, res: express.Response, next: NextFunction): Promise<void | never> => {
+  try {
+    const values = (await extractPayload(req)) as Fields;
+    if (instanceOfCompany(values)) {
+      const { email, orgnum, username, salt, passhash, name }: Company = values;
+      logger.info(`Creating new company ${username}`);
+      const pool: Pool = await genereateAdminConnection();
+      const poolClient: PoolClient = await getPoolClient(pool);
```

Comment:
What is the purpose of these lines, is it to assert that the user is an admin?

Reply:
Those lines get a connection to the database that are admins. 

It is the only way to do "Create role" and "grant" statements.

Reply:
Okay, so they are not used to actually establish that the user has admin rights?

Is there some middleware restricting access to routes which require admin privileges?

Reply:
The next commit i'm adding is the one containing the rights management for express.

So the next PR will include OICD auth using OW4. 

So the short answer is no. These user do not have admin rights, they only need an admin on the db to be created as roles in the db.